app/policies/idv/step_info.rb

```diff
@@ -8,17 +8,28 @@ class StepInfo
     validates :action, presence: true
     validate :next_steps_validation, :preconditions_validation, :undo_step_validation
 
-    def initialize(key:, controller:, next_steps:, preconditions:, undo_step:, action: :show)
+    def initialize(
+      key:,
+      controller:,
+      next_steps:,
+      preconditions:,
+      undo_step:,
+      action: :show
+    )
       @key = key
-      @controller = controller
-      @action = action
+      @controller = controller.name.underscore.gsub('_controller', '')
```

Comment:
Sadly, we can't call the `full_controller_name` class method here because the instance isn't fully initialized yet.